sycl/source/detail/global_handler.cpp

```diff
@@ -212,6 +216,18 @@ void GlobalHandler::drainThreadPool() {
     MHostTaskThreadPool.Inst->drain();
 }
 
+#ifdef _WIN32
+  // because of something not-yet-understood on Windows
+  // threads may be shutdown once the end of main() is reached
+  // making an orderly shutdown difficult. Fortunately, Windows
+  // itself is very aggressive about reclaiming memory. Thus,
+  // we focus solely on unloading the plugins, so as to not
+  // accidentally retain device handles. etc 
+void shutdown(){
+  GlobalHandler *&Handler = GlobalHandler::getInstancePtr();
+  Handler->unloadPlugins();
```

Comment:
Could you please add a comment saying that piTearDown might not be safe to call low level API, since there might be dependent libraries that are unloaded.

Reply:
I added a comment to each definition of piTearDown in the plugin libs.